### ISSUE TITLE
Can't boot properly on Windows 10

### ISSUE THREAD

**Issue by ZeCraum** [supercollider/supercollider#4368]
## Environment

* SuperCollider version: 3.10.2
* Operating system: Windows 10 64bits

When I try to boot it, the server status goes yellow and freeze with the post window at:
`Booting server 'localhost' on address 127.0.0.1:57110.`

To unfreeze it I kill all the servers and then got this message:

```
ÊXITO: o processo "scsynth.exe" com PID 5776 foi finalizado.
Server 'localhost' exited with exit code 1.
```
Translating: Success: the process "scsynth.exe" with PID5776 was finished. 

I need to boot, kill, boot, kill, boot, kill... many times until a moment that it finally properly boots!

After booting properly it display the last part of this message many times:

```
Booting server 'localhost' on address 127.0.0.1:57110.

Device options:
  - MME : Mapeador de som da Microsoft - Input   (device #0 with 2 ins 0 outs)
  - MME : Microphone (High Definition Aud   (device #1 with 2 ins 0 outs)
  - MME : Microphone (High Definition Aud   (device #2 with 2 ins 0 outs)
  - MME : Mapeador de som da Microsoft - Output   (device #3 with 0 ins 2 outs)
  - MME : Speakers (High Definition Audio   (device #4 with 0 ins 2 outs)
  - Windows DirectSound : Driver de captura de som primário   (device #5 with 2 ins 0 outs)
  - Windows DirectSound : Microphone (High Definition Audio Device)   (device #6 with 2 ins 0 outs)
  - Windows DirectSound : Microphone (High Definition Audio Device)   (device #7 with 2 ins 0 outs)
  - Windows DirectSound : Driver de som primário   (device #8 with 0 ins 2 outs)
  - Windows DirectSound : Speakers (High Definition Audio Device)   (device #9 with 0 ins 2 outs)
  - Windows WASAPI : Speakers (High Definition Audio Device)   (device #10 with 0 ins 2 outs)
  - Windows WASAPI : Microphone (High Definition Audio Device)   (device #11 with 2 ins 0 outs)
  - Windows WASAPI : Microphone (High Definition Audio Device)   (device #12 with 2 ins 0 outs)
  - Windows WDM-KS : Microphone 1 (HD Audio Muxed capture)   (device #13 with 2 ins 0 outs)
  - Windows WDM-KS : Microphone 2 (HD Audio Muxed capture)   (device #14 with 2 ins 0 outs)
  - Windows WDM-KS : Speakers (HD Audio Headphone/Speakers)   (device #15 with 0 ins 2 outs)

Booting with:
  In: MME : Microphone (High Definition Aud
  Out: MME : Speakers (High Definition Audio
  Sample rate: 44100.000
  Latency (in/out): 0.013 / 0.091 sec
SC_AudioDriver: sample rate = 44100.000000, driver's block size = 64
SuperCollider 3 server ready.
Requested notification messages from server 'localhost'
localhost: server process's maxLogins (1) matches with my options.
localhost: keeping clientID (0) as confirmed by server process.
Safety('localhost') is running, using 'safeClip'.
Shared memory server interface initialized
server 'localhost' disconnected shared memory interface
Shared memory server interface initialized
server 'localhost' disconnected shared memory interface
Shared memory server interface initialized
server 'localhost' disconnected shared memory interface
Shared memory server interface initialized
server 'localhost' disconnected shared memory interface
Shared memory server interface initialized
server 'localhost' disconnected shared memory interface
Shared memory server interface initialized
```

**Comment by geoffroymontel** [comment]
Hi ! I've experienced the same problem on my Windows 10 PC the first time I'm launching SuperCollider after booting.
But if I'm just waiting for a few minutes when it seems to freeze and display `Booting server 'localhost' on address 127.0.0.1:57110`, it ends up booting.
And then, the next times I am starting it, it goes fast.
I've had the feeling that it's caused by a recent change of the Windows builtin antivirus which scans executables. But I'm not sure.

Could you please try and start SC after a fresh boot, and just wait for a few minutes when it seems to hang, and report your results please ?

**Comment by ZeCraum** [comment]
You are totally right! I was counting here and it took 1min and 30s to boot, after reseting it is now working normally... windows is weird.

**Comment by geoffroymontel** [comment]
Thanks for your confirmation !
I will try to find out if it relates to the Windows antivirus or not.

**Comment by ntrocado** [comment]
I think I have the same problem: fresh boot takes about 1 minute, subsequent ones are normal.
I tried adding an exclusion for scsynth.exe in the built-in antivirus, but unfortunately that didn't work.

**Comment by snappizz** [comment]
while this bug is not critical, the solution is not obvious. until we can find a way to fix this delay, it would be good to print a message for windows users to warn them of this issue.

**Comment by geoffroymontel** [comment]
I am confirming that this is due to Windows Defender.
If you add scide.exe sclang.exe and scsynth.exe process to the Windows Defender exclusion list, server boots immediately.
![Capture d’écran (3)](https://user-images.githubusercontent.com/352910/58980171-064a7100-87d0-11e9-9345-1732542f6447.png)
![Capture d’écran (4)](https://user-images.githubusercontent.com/352910/58980174-08143480-87d0-11e9-9fcb-13458cd9c7cf.png)


.

**Comment by geoffroymontel** [comment]
Maybe we could submit SuperCollider to Microsoft Security Intelligence here : https://www.microsoft.com/en-us/wdsi/filesubmission

**Comment by ntrocado** [comment]
You're right, it is indeed due to Windows Defender. I was adding the /file/ to the exclusion list, but it is the /process/ `scsynth.exe` that must be added, and the problem is solved!

**Comment by geoffroymontel** [comment]
Whitelisting `scsynth.exe` process is enough.

**Comment by dyfer** [comment]
Is there a way to trigger a prompt asking `Do you want to add sclang.exe to Windows Defender exclusion list?` or similar that would guide the user and automate this process?

**Comment by geoffroymontel** [comment]
I hope we can solve this by submitting scsynth to [Microsoft Security Intelligence team](https://www.microsoft.com/en-us/wdsi/filesubmission).
But we could display a warning message telling the user to add scsynth.exe to Windows Defender exclusion list, yes. 
Automating the process might be trickier

**Comment by snappizz** [comment]
anyone who can reproduce this issue: we suspect it's related to #4496. can you try this before booting the server?

> Server.program = "scsynth -B 127.0.0.1";

**Comment by geoffroymontel** [comment]
I've just tried adding this line, and the issue remains, I get this line in the post window
```
Booting server 'localhost' on address 127.0.0.1:57110.
```
but the server takes more than a minute to boot.

**Comment by dyfer** [comment]
> You're right, it is indeed due to Windows Defender. I was adding the _file_ to the exclusion list, but it is the _process_ `scsynth.exe` that must be added, and the problem is solved!

I can confirm that this fixed long boot for me as well on Windows 10.

**Comment by geoffroymontel** [comment]
For the record, more info about Windows Defender "block at first sight" "feature"
https://docs.microsoft.com/fr-fr/windows/security/threat-protection/windows-defender-antivirus/configure-block-at-first-sight-windows-defender-antivirus
https://docs.microsoft.com/en-us/windows/security/threat-protection/windows-defender-antivirus/configure-block-at-first-sight-windows-defender-antivirus

**Comment by snappizz** [comment]
thanks for the info @geoffroymontel! i had a look at the docs and it seems like the best option is to submit a file to Microsoft for analysis so they can help us understand what's wrong: https://docs.microsoft.com/en-us/windows/security/threat-protection/intelligence/submission-guide

here is the submission page: https://www.microsoft.com/en-us/wdsi/filesubmission/

it seems that microsoft prioritizes submissions by volume, so it may be good if everyone affected submits scsynth.exe for inspection as "home consumer". (@geoffroymontel you can submit as "software developer")

**Comment by geoffroymontel** [comment]
I used to have a developer account a few years ago, I will try and submit scsynth.exe and keep you posted !

**Comment by geoffroymontel** [comment]
I've just submitted scsynth.exe from SC 3.10.3, will keep you updated !

**Comment by geoffroymontel** [comment]
Mmm, first response from Microsoft

```
Submission ID: a4ccd87a-09f0-4bff-9b39-72373382c954
Status: Completed
Submitted by: g***@***.com
Submitted: Sep 4, 2019 07:55:58
User Opinion: Incorrect detection
Analyst comments:
                The file is not malware and we cannot reproduce any detection on the file.  If detection is still observed, please follow the steps below to capture support log files from the system reporting detection.

On Windows 10, from elevated command prompt, change to directory "%programfiles%\windows defender" and execute mpcmdrun.exe with option GetFiles:
    cd "%programfiles%\windows defender"
    mpcmdrun.exe -GetFiles

On Windows 7, from elevated command prompt, change to directory "%programfiles%\microsoft security client" and execute mpcmdrun.exe with option GetFiles:
    cd  "%programfiles%\microsoft security client"
```

Will keep you posted.

**Comment by geoffroymontel** [comment]
I provided more logs to Microsoft and here is their answer

```
                Thank you providing logs.
We have gone through support files and we have not seen any detection for the file submitted in submission "a4ccd87a-09f0-4bff-9b39-72373382c954".
We also tried to replicate scenario at our end but uinfortunately we are not able to reproduce it.
At present we have marked the file in submisison "a4ccd87a-09f0-4bff-9b39-72373382c954" as clean and expecting issue to be resolved.
Kindly cross check the issue once again with latest signature. If issue persist please provide us with steps to reproduce issue at our end to diagnose it further and to provide relevant solution.

Thank you for contacting Microsoft.
```

Will let you know tonight if it seems to fix the startup delay problem... or not !

**Comment by joshpar** [comment]
Hi @geoffroymontel - any updates on this? Thanks!

**Comment by geoffroymontel** [comment]
Unfortunately it did not solve the problem. I'm back, I will keep up with this with Microsoft right now.

**Comment by geoffroymontel** [comment]
Submitted another ticket today, will keep you updated.

**Comment by geoffroymontel** [comment]
Arg, after like 10 back and forth with submission portal, I guess they can't help us.
Trying [Stack Overflow](https://stackoverflow.com/questions/58383531/why-is-windows-defender-delaying-the-start-of-our-piece-of-software)

**Comment by geoffroymontel** [comment]
Got an answer on my Stack Overflow question. I guess the next step is to install Visual Studio on a Windows PC, and try to trace where it's hanging.
I'm a bit reluctant to install Visual Studio on my Windows PC which is dedicated to music... Does anybody here can reproduce the problem on its PC and has Visual Studio setup ?

**Comment by snappizz** [comment]
@geoffroymontel thanks for your continued efforts. in case we get stuck on this, feel free to broadcast a "help wanted" signal to sc-users and scsynth.org.

**Comment by snappizz** [comment]
@jrsurge pointed out at the last dev meeting that everyone who has reported this issue seems to have Windows system language set to something other than English. might provide some clues.

**Comment by geoffroymontel** [comment]
Help wanted posted in scsynth.org ! https://scsynth.org/t/looking-for-experienced-windows-developers-to-tackle-an-issue/1491

**Comment by jrsurge** [comment]
I noticed this issue in a different context.

I'm going to try to get hold of the machine in question and investigate, but as an initial report in case it gives any more clues:

## Machine A
- Win 10 Education (1809)
- SuperCollider 5.10.3
- Soundcard: RME MADIface XT
- API: ASIO

## Machine B
- Macbook Air (I think)
- macOS (no idea)
- API: CoreAudio

## Steps to reproduce
- Machine A:
  - Start scide, booting scsynth (no issues)
  - Close scide (kills server)
  - Unplug soundcard
- Machine B:
  - Plug soundcard into alt computer
  - Unplug soundcard from alt computer
- Machine A:
  - Start scide, booting scsynth
  - Replug soundcard into main computer
  - scsynth pauses for ~60 seconds.

This happened every time we swapped machines - so clean system boots don't seem to be a part of it.
It was during concerts, so I didn't have any time to figure anything out.

I'll try to investigate further, it's just a difficult computer to access.

**Comment by geoffroymontel** [comment]
Going back to this.
1. I created a Windows VM on my Mac and could not reproduce the issue with 3.10.4
2. Going back to my Windows machine where I could reproduce it, deleting startup file, Extensions folder and downloaded-quarks folder, can't reproduce either.
3. putting back startup file -> still can't reproduce
4. putting back SC3 Plugins -> now I can reproduce the problem

So it's something with SC3 Plugins

Next step : going through all SC3 Plugins to try and find the one to blame !

**Comment by dyfer** [comment]
Someone commented on LadspaUGen triggering antivirus alert - maybe that would be the one to start with?

**Comment by geoffroymontel** [comment]
Let me check right now

**Comment by geoffroymontel** [comment]
MMmmmm
1. removing LadspaUgen folder, rebooting -> can't reproduce
2. putting back LadspaUgen folder, rebooting -> can't reproduce
3. removing SC3 Plugins, downloading latest SC3 Plugins, installing them -> can't reproduce

This is confusing.

**Comment by geoffroymontel** [comment]
OK. I can reproduce the problem with a Windows 10 VM on VirtualBox on my Mac, with SC 3.10.4 and SC3-Plugins. Will continue to dig into it.

**Comment by geoffroymontel** [comment]
Made more tests with my Windows machine and the VM and I am not getting anywhere.
I can definitely reproduce the problem but not consistently. I can't find a perfect reproducer...

**Comment by geoffroymontel** [comment]
I could reproduce the problem at every reboot, now I can't. It seems Windows Defender delays scsynth starttup after reboot but only after a few hours/days. Something has changed with newer Windows 10 updates.
At least we are not the sole developers to face this problem : https://answers.microsoft.com/en-us/protect/forum/protect_defender-protect_updating-windows_10/last-update-of-windows-defender-causes-terribly/c71cc133-5c5f-447e-95fb-e2641085836d

**Comment by brianlheim** [comment]
it sounds like this is a Windows issue then, and that they already have a reproducer. so nothing to do but wait until they fix it upstream.

**Comment by totalgee** [comment]
I'm glad I came across the info in this issue...this is something that has long plagued me. I'm glad to discover that I can reduce that arbitrarily-long "first-time" boot delay by adding scsynth.exe as a Windows Defender exclusion! Hopefully MSFT will solve this on their end, but for now it's good to know the workaround... Thanks.

**Comment by geoffroymontel** [comment]
Update on this issue : there's a PR ready to display a warning message if the server takes too long to start, but @jrsurge is also investigating for a better fix.

For people here who also experience this problem : can you tell us if you get the scsynth boot delay if you don't have sc3-plugins installed ? I don't.

**Comment by totalgee** [comment]
On one of my machines, the server never seems to boot, unless I temporarily disable "virus and threat detection" in Windows Defender (now called Windows Security). On that machine, I **have** set scsynth.exe, sclang.exe and scide.exe as exclusions, but it still doesn't work (on my other machines that exclusion setting seems to help).

Along with sc3-plugins, could it also be related to specific Quarks? (e.g. the VSTPlugin Quark, which may scan the disk for VST plugins, and that probably looks suspicious to Windows Security -- although this should happen after scsynth is booted?).

**Comment by totalgee** [comment]
@geoffroymontel Indeed, I just tried getting rid of sc3-plugins temporarily, and it booted just fine and immediately (perhaps because of my exclusions, perhaps it would have worked anyhow). Maybe I'd have to exclude a *whole* bunch of binary plugins, for it to work with sc3-plugins? (-;

**Comment by geoffroymontel** [comment]
@totalgee did you set up scsynth.exe *process* in the exclusion list ? 
![Capture d’écran (25)](https://user-images.githubusercontent.com/352910/82545407-ef269f80-9b56-11ea-87e6-87137c974c3c.png)
It does not work for me if I put the *file* in the exclusion list, not the process.

**Comment by totalgee** [comment]
@geoffroymontel Ah, **brilliant** -- thanks, yes, that does the trick! I had indeed used "File" type of exclusion for the SC executables on disk. Seems to boot fine (immediately) if I put `scsynth.exe` as a *process* to exclude.

**Comment by schollz** [comment]
I could use some help here - supercollider was working great last week and now it won't boot. 

I've got supercollider binaries in the Windows Defender "Allowed apps":

![image](https://user-images.githubusercontent.com/6550035/116944363-3869e500-ac2a-11eb-9952-a868a417f8b6.png)

I've added exclusions in the Antivirus for anything I could think of:

![image](https://user-images.githubusercontent.com/6550035/116944400-4b7cb500-ac2a-11eb-8d63-955224ab5567.png)

But I keep on getting the message:

```
*** Welcome to SuperCollider 3.11.2. *** For help press Ctrl-D.
SCDoc: Indexing help-files...
SCDoc: Indexed 1902 documents in 1.21 seconds
Booting server 'localhost' on address 127.0.0.1:57110.
Server: possible boot delay.
On some Windows-based machines, Windows Defender sometimes delays server boot by one minute.
You can add scsynth.exe and/or supernova.exe *processes* to Windows Defender exclusion list to avoid this delay. It's safe.
```

Any help would be greatly appreciated :)

**Comment by joshpar** [comment]
Thanks @schollz - did anything happen over the past week? System update, SC update, or anything like that?

**Comment by schollz** [comment]
@joshpar Okay yeah it looks like there was a System update last week:

![image](https://user-images.githubusercontent.com/6550035/116945786-58e76e80-ac2d-11eb-87bd-95c1e5f4b928.png)

Once it stopped working I tried uninstalling supercollider (3.11.0) and then updating to the latest version (now 3.11.2) and I redid all the exclusions (as seen above).

**Comment by dyfer** [comment]
Sorry for an obvious question @schollz  - have you waited for an extended period of time (2+ mins) to see if the server boots eventually? If it does eventually boot, the subsequent boots should be quick (i.e. regular).

**Comment by schollz** [comment]
@dyfer Great question :) I've waited 2+ minutes before and no boot. Just in case, I am going to leave it on now for hours+ to see if anything happens.

And I really appreciate the support, thank you for your help.

**Comment by geoffroymontel** [comment]
Hi ! Installing the same Windows update to see how it's going on my PC.

**Comment by alln4tural** [comment]
Note it says *process* in the screenshot you provided. As i dimly recall, that is a different setting than "app".On 04/05/2021, 02:08 Zack ***@***.***> wrote:

   
  @dyfer Great question :) I've waited 2+ minutes before and no boot. Just in case, I am going to leave it on now for hours+ to see if anything happens. 
  And I really appreciate the support, thank you for your help.

**Comment by geoffroymontel** [comment]
OK. I installed all the updates including KB4023057 and the server is booting fine on my PC.
@schollz could you try removing all the Windows Defender exclusions except `scsynth.exe` process ?
I checked my Windows Defender Firewall settings and I only have sclang in there, on private networks.

**Comment by totalgee** [comment]
Yes, that's also what I found was needed. The important detail is the
**process** type of exclusion, and only needed for scsynth.exe (no need for
full path, so the same exclusion works with different install locations).

**Comment by schollz** [comment]
Thanks everyone, really appreciate this help. Unfortunately I'm still stuck with the problem. I tried removing everything from my exclusion list except for scsynth.exe (Process) and still no go.

![image](https://user-images.githubusercontent.com/6550035/117009810-fdef5f00-aca0-11eb-8e96-a7347ccfb2bc.png)

I tried just turning off all the antivirus stuff (Real-time protection off, Cloud-delivered protection off, Automatic sample submission off, Tamper protection off) and turning off the firewall (public and private) completely and also no go, even after waiting for a few minutes.

![image](https://user-images.githubusercontent.com/6550035/117010019-3b53ec80-aca1-11eb-9fc6-f1bf5aae89bc.png)

**Comment by geoffroymontel** [comment]
Mmm, weird. Can you post the content of your `startup.scd` file ?

**Comment by schollz** [comment]
@geoffroymontel It looks like my `startup.scd` file is completely empty. Is that a problem?

![image](https://user-images.githubusercontent.com/6550035/117014155-60e2f500-aca5-11eb-85a6-234ad27811fa.png)

**Comment by geoffroymontel** [comment]
Nope, it's not a problem at all. I'm clueless... I would try uninstalling all quarks and server extensions (if any) to see if it helps...

**Comment by geoffroymontel** [comment]
Mmmm. I guess you already tried rebooting your PC... You could also try starting scsynth from the command line to see how it goes.

**Comment by schollz** [comment]
I did a couple restarts to no avail. Looks like the command line is also stuck:

```
> C:\..\SuperCollider-3.11.2> .\scsynth.exe -t 57110
Server: possible boot delay.
On some Windows-based machines, Windows Defender sometimes delays server boot by one minute.
You can add scsynth.exe and/or supernova.exe *processes* to Windows Defender exclusion list to avoid this delay. It's safe.
```

**Comment by geoffroymontel** [comment]
😞 I'm sorry I can't think of any other thing to look after right now...

**Comment by dyfer** [comment]
I am able to reproduce this after running Windows Update yesterday... No way to start scsynth 64-bit (!!!) I tried from command line as well, same thing.

Also, I _am_ able to start 32-bit scsynth (which prints errors loading plugins since I haven't removed my 64-bit sc3-plugins, but it does start...)

**Comment by schollz** [comment]
> Also, I am able to start 32-bit scsynth

Okay this also works for me. I uninstalled 64-bit and installed 32-bit and its working again as 32-bit :)

**Comment by martinhurych** [comment]
Hello, I have the same problem with booting of sc server on Windows. I've found that it's caused by the latest Microsoft Visual C+ 2015-2019. If it's uninstalled then the sc server is booting properly. Does anybody know how to preserve the Visual C+ installation and at the same time boot the sc server?

**Comment by totalgee** [comment]
I use VS 2019 (and before that 2017) and have no problems booting the SC server (once the above-mentioned [process exclusion](https://github.com/supercollider/supercollider/issues/4368#issuecomment-831754734) is added to the Windows Security/Defender antivirus settings). Definitely works for me (on the latest Windows 10 Pro and earlier versions) with SC 3.11.1 and 3.11.2.

Also, sclang is allowed to communicate through the Windows Defender Firewall on my private (local) network.

**Comment by totalgee** [comment]
And for the record, I'm running the 64-bit version of SC:
```
$ file scsynth.exe
scsynth.exe: PE32+ executable (console) x86-64, for MS Windows
```

**Comment by martinhurych** [comment]
I've tried it out. The process exclusion doesn't help. Only the uninstallation of VS 2019 helps for running sc x64, but then the other programs don't work. Anyway, the x32 bit version of sc works also with VS 2019.

**Comment by totalgee** [comment]
Does the server program actually run (e.g. if you run it from the command line)?

```
$ scsynth -u 57110

[...]
Booting with:
  In: Windows DirectSound : Primary Sound Capture Driver
  Out: Windows DirectSound : Primary Sound Driver
  Sample rate: 44100.000
  Latency (in/out): 0.001 / 0.121 sec
SC_AudioDriver: sample rate = 44100.000000, driver's block size = 64
SuperCollider 3 server ready.
```

If it does run and show "ready", then (probably?) the issue is the (UDP) communication between sclang and scsynth...which suggests it's the firewall that is blocking things. If scsynth doesn't start at all, then I don't know what to suggest...

**Comment by schollz** [comment]
> In my case the problem was Jack. After uninstalling it SC boots fine

Thank you so much @giohappy. Removing JACK also fixed my problem. I can run the 64-bit SuperCollider (now on 3.12.2).

**Comment by dyfer** [comment]
Just to reiterate - if the server doesn't run at all, then it's likely an issue with an _old_ version of Jack. Running current Jack (>1.9.18) should fix this.

Let's keep this thread for issues with _boot delay_.

**Comment by totalgee** [comment]
Update: I did a clean Windows 10 Pro install (21H1, 10.0.19043) on an old laptop today, and installed the latest SC 3.12.2 (64-bit). I was able to boot the SC server without any problems or delays on a clean install. So far, so good. (I didn't add the Process exclusion for `scsynth.exe` and it still boots immediately).

@geoffroymontel I wonder if it's possible Microsoft has finally flagged it as "safe"?

**Comment by totalgee** [comment]
> @totalgee are you booting your server in `startup.scd` or just setting the audio device?

No, just setting audio device in Server options. When I say "delay" I mean from whenever I do ctrl-b (or `s.boot`).

> Also, I wonder if boot delay is at all connected to which driver (MME vs ASIO) is being used.

It seems to delay when I use an audio interface (e.g. Focusrite), whether with ASIO or another API like DirectSound or WASAPI. (I never use MME, nor should anyone 😉).

**Comment by dyfer** [comment]
Circling back to this: in context of #6774, we may want to:
- revise the wording of the message printed; in particular, searching for "Windows Defender" doesn't seem to be the best way to find this setting in Windows 11
- add documentation to the Server help with a tiny bit more information how to add an exception (i.e. Windows Security -> Virus and Threat protection -> Manage settings -> Exclusions -> Add or remove exclusion -> Add an exclusion -> Process -> process name: "scsynth.exe"). The boot delay message should also point to that help document

**Comment by prko** [comment]
In the last three days, I tested some machines:
- Windows 11 on intel 7 
- Windows 10 on intel i3
- Windows 10 on intel i5(?)
- Windows 11 via parallels desktop on imac 2013
- Windows 10 via parallels desktop on imac 2013

The following warning and server boot delay seem to occur only on Windows 11 on virtual machines (arm64 and Intel):

>On some Windows-based machines, Windows Defender sometimes delays server boot by one minute.
You can add scsynth.exe and/or supernova.exe processes to the Windows Defender exclusion list to avoid this delay. It's safe.

**Comment by prko** [comment]
This is again extremely when using [0f4fd5e]. To boot the local server, I need more than several minutes. With this, sc cannot be used for the students who use Windows.

**Comment by dyfer** [comment]
@prko we couldn't narrow down the issue. It doesn't seem to be directly related to anything we control.

> I need more than several minutes. With this, sc cannot be used for the students who use Windows.

I thought that after the first delayed boot, the subsequent boots are fine. Is that not the case?

Also, a workaround for this is to add scsynth process to the exclusion list: Windows Security -> Virus and Threat protection -> Manage settings -> Exclusions -> Add or remove exclusion -> Add an exclusion -> Process -> process name: "scsynth.exe". But again, even without doing this after the first long boot the subsequent boots should not be delayed.

**Comment by prko** [comment]
Out of around 15 students using Windows, at least two have encountered this issue.  
This seems to be a notable obstacle for Windows users.  
It’s unfortunate that there doesn’t seem to be a clear solution at the moment....
